.github/workflows/ci.yaml

```diff
@@ -0,0 +1,24 @@
+name: CI
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+# Automatically cancel in-progress actions on the same branch
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v4
+        with:
+          go-version: "1.20"
+      - uses: actions/checkout@v3
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: latest
```

Comment:
A little extra context on these GitHub Actions configs: I personally wouldn’t recommend spending time learning this (for now). Most projects you work on will already have this set up for you. And for these projects, once it’s set up correctly, these files rarely change.

If you ever have to set these up in the future, 99% of the time you’re going to start by copying a setup from a project that already does what you want (it’s what I did here—just copy from another Go project I’ve worked on).

So again, don’t worry about this file. Only pay attention to the actual **logs** it generates (click on the ❌ to see why it failed, and in all PRs after this, make sure it passes!)